Revert https://github.com/numpy/numpy/commit/eb6be7c4765665724cd12431bfefb050ba0f2d4b.

See also:
https://github.com/pypa/setuptools/issues/2372
https://github.com/numpy/numpy/issues/20692

In the ebuild, we're forcing SETUPTOOLS_USE_DISTUTILS=stdlib which uses
the distutils version from within Python (which will be removed in 3.11)
rather than the bundled-in-setuptools-60 version which breaks numpy.
--- a/setup.py
+++ b/setup.py
@@ -80,10 +80,6 @@ if os.path.exists('MANIFEST'):
 # so that it is in sys.modules
 import numpy.distutils.command.sdist
 import setuptools
-if int(setuptools.__version__.split('.')[0]) >= 60:
-    raise RuntimeError(
-        "Setuptools version is '{}', version < '60.0.0' is required. "
-        "See pyproject.toml".format(setuptools.__version__))
 
 # Initialize cmdclass from versioneer
 from numpy.distutils.core import numpy_cmdclass
